### ISSUE TITLE
Aggregators 'max' and 'min' can be applied to booleans, causing failure

### ISSUE THREAD

**Issue by hesselink** [tomjaguarpaw/haskell-opaleye#145]
The aggregation functions `max` and `min` in opaleye have a `PGOrd` constraint, and an instance for `PGBool` exists. However, postgres doesn't allow `max` and `min` on booleans. The allowed argument types are ["any array, numeric, string, or date/time type"](http://www.postgresql.org/docs/9.4/static/functions-aggregate.html). Removing the instance isn't the solution, of course, since things like `.>` do work on booleans.

**Comment by tomjaguarpaw** [comment]
That's a rather unnecessarily painful restriction on the part of Postgres.  Oh well.  I guess we'll have to have two typeclasses.

**Comment by ocharles** [comment]
I wonder if it's not easier to just add the `max` function to your PostgreSQL database.

**Comment by hesselink** [comment]
Wouldn't that mean that everyone using Opaleye would have to add this to avoid these errors?

**Comment by ocharles** [comment]
Yes, if they rely on that, so it is a little unfortunate. I'm just wondering if it's really worth Opaleye's time capturing all the inconsistencies. I guess it is ultimately safer though... Opaleye could come with a prelude that should be installed, but I agree it's not a particularly compelling argument. It also might be worth fixing upstream directly in PostgreSQL.

**Comment by xldenis** [comment]
opaleye should be able to interface with existing psql databases without forcing changes on them so that opaleye can guarantee query correctness. It seems like psql has a lot of these kinds of restrictions on input types for functions. It may be worth building a table of all the 'stdlib' functions and their input / output mappings to see if there are large sets that could be grouped into typeclasses.
